applications/GeoMechanicsApplication/tests/cpp_tests/test_utilities/spy_element.cpp

```diff
@@ -0,0 +1,78 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+#include "spy_element.h"
+
+namespace Kratos::Testing {
+
+void SpyElement::FinalizeSolutionStep(const ProcessInfo& rCurrentProcessInfo)
+{
+    mSolutionStepFinalized = true;
+}
+
+void SpyElement::InitializeSolutionStep(const ProcessInfo& rCurrentProcessInfo)
+{
+    mSolutionStepInitialized = true;
+}
```

Comment:
I would swap these two implementations (initialize first, then finalize).

Also, I would remove the function parameter names, since they are not used:
```suggestion
void SpyElement::FinalizeSolutionStep(const ProcessInfo&)
{
    mSolutionStepFinalized = true;
}

void SpyElement::InitializeSolutionStep(const ProcessInfo&)
{
    mSolutionStepInitialized = true;
}
```
(This suggestion also applies to the other member functions.)

Reply:
Done